src/Find_monomial.jl

```diff
@@ -1,54 +1,77 @@
-function find_monomial(mon1::Monomial{C}, mon2::PolyVar{C}) where {C}
-    return findall( x -> x == mon2, variables(mon1))
-end
-
-function _subvectors(v::AbstractVector, l::Int)
-    @assert length(v) >= l && l > 0
-    return [v[i:i+l-1] for i = 1:length(v)-l+1]
-end
-
 """
-    unzero_termlike(term::MTermLike)
+    findfirst(mon1::Monomial{C}, mon2::MMonomialLike{C})
 
-Returns a Term or a Monomial where all the variables()[i] with exponents()[i]==0 are eliminated
-thus a Term or a Monomial without zeros in the exponents field.
+Returns first position where mon2 can be found in the representation of mon1.
+Second output is the length of mon2.
 """
-function unzero_termlike(ter::Term{C, T}) where {C, T}
-    return Term{C,T}(coefficient(ter), unzero_term(monomial(ter)));
+function Base.findfirst(mon1::MMonomialLike{C}, mon2::PolyVar{C}) where {C}
+        position = findfirst( x -> x == mon2, variables(mon1))
+        if position isa Nothing 
+            return 0, 1
+        else
+            return position, 1
+        end
 end
-function unzero_termlike(mon::Monomial{C}) where {C}
-    f = findall(x->x!=0, exponents(mon));
-    if isempty(f)
-        return 1;
-    end
-    z       = Int[];
-    var     = PolyVar{C}[];
-    i       = 0;
-    for p in f
-        push!(z,exponents(mon)[p]);
-        push!(var, variables(mon)[p]);
+
+function _subvectors(v::AbstractVector, l::Int)
+    if l == 0
+        return []
+    else
+        @assert length(v) >= l 
+        return [v[i:i+l-1] for i = 1:length(v)-l+1]
     end
+end
 
-    return Monomial{C}(var, z);
+function _match(c::Bool, v1::AbstractVector, v2::AbstractVector)
+    if c || length(c) <= 2
+        return all(v1-v2.>=0)
+    else
+        v = v1-v2
+        return first(v) >= 0 && last(v) >= 0 && all(v[2:end-1] .>=0)
```

Comment:
The Boolean c is to determine whether the variables are commuting (c==true) or not (c==false). 
If they are commuting we just need >= for all entries to be able to replace the monomial. 
In case they are not, the middle part needs to be equal.